pkg/virtctl/memorydump/memorydump.go

```diff
@@ -0,0 +1,367 @@
+	}
+	if accessMode != "" && accessMode != string(k8sv1.ReadWriteOnce) && accessMode != string(k8sv1.ReadWriteMany) {
+		return nil, fmt.Errorf("invalid access mode, use either ReadWriteOnce or ReadWriteMany if supported")
+	}
+	if accessMode != "" {
+		pvc.Spec.AccessModes = []k8sv1.PersistentVolumeAccessMode{k8sv1.PersistentVolumeAccessMode(accessMode)}
+	} else {
+		pvc.Spec.AccessModes = []k8sv1.PersistentVolumeAccessMode{k8sv1.ReadWriteOnce}
+	}
+
+	return pvc, nil
+}
+
+func checkNoExistingPVC(namespace, claimName string, virtClient kubecli.KubevirtClient) error {
+	_, err := virtClient.CoreV1().PersistentVolumeClaims(namespace).Get(context.Background(), claimName, metav1.GetOptions{})
+	if err == nil {
+		return fmt.Errorf("PVC %s/%s already exists, check if it should be created if not remove create flag", namespace, claimName)
+	}
+	if !k8serrors.IsNotFound(err) {
+		return err
+	}
+	return nil
+}
+
+func checkNoAssociatedMemoryDump(namespace, vmName string, virtClient kubecli.KubevirtClient) error {
+	vm, err := virtClient.VirtualMachine(namespace).Get(vmName, &metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	if vm.Status.MemoryDumpRequest != nil {
+		return fmt.Errorf("please remove current memory dump association before creating a new claim for a new memory dump")
+	}
+	return nil
+}
+
+func createPVCforMemoryDump(namespace, vmName, claimName string, virtClient kubecli.KubevirtClient) error {
+	// Before creating a new pvc check that there is not already
+	// assocaited memory dump pvc
+	if err := checkNoAssociatedMemoryDump(namespace, vmName, virtClient); err != nil {
+		return err
+	}
+
+	if err := checkNoExistingPVC(namespace, claimName, virtClient); err != nil {
+		return err
+	}
+
+	memoryDumpExpectedSize, err := calcMemoryDumpExpectedSize(vmName, namespace, virtClient)
+	if err != nil {
+		return err
+	}
+
+	neededSize, err := calcPVCNeededSize(memoryDumpExpectedSize, &storageClass, virtClient)
+	if err != nil {
+		return err
+	}
+
+	pvc, err := generatePVC(neededSize, claimName, namespace, storageClass, accessMode)
+	if err != nil {
+		return err
+	}
+
+	_, err = virtClient.CoreV1().PersistentVolumeClaims(namespace).Create(context.Background(), pvc, metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("PVC %s/%s created\n", namespace, claimName)
+
+	return nil
+}
+
+func createMemoryDump(namespace, vmName, claimName string, virtClient kubecli.KubevirtClient) error {
+	memoryDumpRequest := &v1.VirtualMachineMemoryDumpRequest{
+		ClaimName: claimName,
+	}
+
+	err := virtClient.VirtualMachine(namespace).MemoryDump(vmName, memoryDumpRequest)
+	if err != nil {
+		return fmt.Errorf("error dumping vm memory, %v", err)
+	}
+	fmt.Printf("Successfully submitted memory dump request of VM %s\n", vmName)
+	return nil
+}
+
+func getMemoryDump(namespace, vmName string, virtClient kubecli.KubevirtClient) error {
+	if createClaim {
+		if claimName == "" {
```

Comment:
Shouldn't we check if this is empty also when `!createClaim`?

Reply:
no because we allow the user to do another get command to the already associated pvc without the need to state the name again